src/ringct/rctTypes.cpp

```diff
@@ -251,7 +251,8 @@ namespace rct {
         }
     }
 
-    size_t n_bulletproof_amounts(const Bulletproof &proof)
+    template<typename proof_t>
+    static size_t n_bulletproof_bulletproof_plus_amounts(const proof_t &proof)
```

Comment:
If you are going to do this, then you need to assert that `BULLETPROOF_MAX_OUTPUTS == BULLETPROOF_PLUS_MAX_OUTPUTS`, and maybe static assert that `proof_t` is `BulletProof` or `BulletProofPlus`. My recommended function signature was meant to decouple the function from the implementation details.

Reply:
Good point on the first one.